src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -540,8 +541,31 @@ public uint GetPaginationLimit(int? first)
     /// <summary>
     /// Retrieve the value of LogLevel from runtime
     /// </summary>
-    public Level? LogLevelValue()
+    public ExtendedLogLevel? IsLogLevel()
     {
         return Runtime?.LogLevel?.Value;
     }
+
+    /// <summary>
+    /// Takes in the RuntimeConfig object and checks the LogLevel.
+    /// If LogLevel is not null, it will return the current value as a LogLevel,
+    /// else it will take the default option by checking host mode.
+    /// If host mode is Development, return `LogLevel.Debug`, else
+    /// for production returns `LogLevel.Error`.
+    /// </summary>
+    public static LogLevel GetConfiguredLogLevel(RuntimeConfig runtimeConfig)
+    {
+        ExtendedLogLevel? value = runtimeConfig.IsLogLevel();
+        if (value is not null)
+        {
+            return (LogLevel)value;
```

Comment:
pending answers to previous feedback threads: if there is no difference between Microsoft.Extensions.Logging.LogLevel enum and ExtendedLog level... we should just be saving Microsoft.Extensions.Logging.LogLevel  to the runtimeconfig and that would negate the need for the casting here. 

If the need for the new enum is just to support "null" , see the example for `EntityCacheOptions` where properties are nullable and there is also a property which tracks whether value is user-defined: https://github.com/Azure/data-api-builder/blob/main/src/Config/ObjectModel/EntityCacheOptions.cs

Reply:
Deleted `ExtendedLogLevel`, although the casting is still needed as it is going from `LogLevel?` to `LogLevel`